node/src/accountant/mod.rs

```diff
@@ -2078,208 +2576,303 @@ mod tests {
         let system = System::new(test_name);
         SystemKillerActor::new(Duration::from_secs(10)).start(); // a safety net for GitHub Actions
-        let consuming_wallet = make_paying_wallet(b"consuming");
-        let pending_payable_scanner = ScannerMock::new()
-            .begin_scan_params(&begin_scan_params_arc)
-            .begin_scan_result(Err(BeginScanError::NothingToProcess))
-            .begin_scan_result(Ok(RequestTransactionReceipts {
-                pending_payable: vec![],
+        let receivable_scanner = ScannerMock::new()
+            .scan_started_at_result(None)
+            .scan_started_at_result(None)
+            .start_scan_params(&start_scan_params_arc)
+            .start_scan_result(Err(StartScanError::NothingToProcess))
+            .start_scan_result(Ok(RetrieveTransactions {
+                recipient: make_wallet("some_recipient"),
                 response_skeleton_opt: None,
             }))
             .stop_the_system_after_last_msg();
-        let mut config = make_bc_with_defaults();
+        let earning_wallet = make_wallet("earning");
+        let mut config = bc_from_earning_wallet(earning_wallet.clone());
         config.scan_intervals_opt = Some(ScanIntervals {
             payable_scan_interval: Duration::from_secs(100),
-            receivable_scan_interval: Duration::from_secs(100),
-            pending_payable_scan_interval: Duration::from_millis(98),
+            pending_payable_scan_interval: Duration::from_secs(10),
+            receivable_scan_interval: Duration::from_millis(99),
         });
         let mut subject = AccountantBuilder::default()
-            .consuming_wallet(consuming_wallet.clone())
             .bootstrapper_config(config)
             .logger(Logger::new(test_name))
             .build();
-        subject.scanners.payable = Box::new(NullScanner::new()); //skipping
-        subject.scanners.pending_payable = Box::new(pending_payable_scanner);
-        subject.scanners.receivable = Box::new(NullScanner::new()); //skipping
-        subject.scan_schedulers.update_scheduler(
-            ScanType::PendingPayables,
-            Some(Box::new(
-                NotifyLaterHandleMock::default()
-                    .notify_later_params(&notify_later_pending_payable_params_arc)
-                    .capture_msg_and_let_it_fly_on(),
-            )),
-            None,
+        // Important. Preventing the possibly endless sequence of
+        // PendingPayableScanner -> NewPayableScanner -> NewPayableScanner...
+        subject.scan_schedulers.payable.new_payable_notify = Box::new(NotifyHandleMock::default());
+        subject
+            .scanners
+            .replace_scanner(ScannerReplacement::PendingPayable(ReplacementType::Null));
+        subject
+            .scanners
+            .replace_scanner(ScannerReplacement::Receivable(ReplacementType::Mock(
+                receivable_scanner,
+            )));
```

Comment:
A suggestion from your side: If you use `ScanForReceivables` instead of `StartMessage` you won't need to write the lines 2859...2869.